### ISSUE TITLE
`kitty @ new-window --cwd ...` behaves strangely when using fish shell

### ISSUE THREAD

**Issue by ronjouch** [kovidgoyal/kitty#1348]
Here's me in `$HOME` typing `kitty @ new-window --cwd /opt/kitty` (with `/opt/kitty` containing a git clone of the kitty repo) under two kitties:

1. Kitty with a 2-lines kitty.conf allowing remote control and specifying **`shell bash`**:  
![kitty-newwindow-bash](https://user-images.githubusercontent.com/522085/51953869-0e526880-240c-11e9-9a6a-b0543b0a0cb3.gif)
  
2. Kitty with a 2-lines kitty.conf allowing remote control and specifying **`shell fish`**:  
![kitty-newwindow-fish](https://user-images.githubusercontent.com/522085/51953875-127e8600-240c-11e9-8750-a267871fbe9b.gif)

As you see, **with fish, the `--cwd` argument behaves strangely**:

- Current directory is changed to the eyes of `ls`(printing the correct contents of `/opt/fish`) and my git status shell line (saying `(master|✔)`)
- **But** both `pwd` and `echo $PWD` incorrectly return `~`.

I'm running a freshly-compiled kitty 0.13.1 HEAD @ 12e269e7911c895ae54abf2c270dfa0c154423a9, and fish shell 3.0.0, under Linux. `kitty @ new-window` clearly used to work a few days/weeks ago, as I'm using it in a script arranging my kitty workspace at work, where I've been using kitty+fish for months.

What's going on? Does it look like a misconfiguration of mine? In kitty or fish? Or can it be a kitty changed behavior / bug? Any ideas to dig in?

Thanks for kitty!

**Comment by kovidgoyal** [comment]
I cannot replicate using kitty from master 4 and fish 3.0.0. Steps I tried:

1) Run kitty -o allow_remote_control=y -o shell=fish
2) Run kitty @ new-window --cwd=/somewhere/else
3) Run pwd in the new window
4) /somewhere/else is printed out

So it looks like something in your fish config.

**Comment by ronjouch** [comment]
> So it looks like something in your fish config.

@kovidgoyal weird; can reproduce the bug even with an empty fish config. Okay, thanks for looking at it, quick as usual 👍. Working around the issue by stopping using `--cwd foo` and adding some `kitty @ send-text --match "id:$myWindow" 'cd foo\r'`

**Comment by callahad** [comment]
I'm also experiencing this in fish 3.0.1 with an empty Fish config, so *something* is up beyond just @ronjouch's config. :)

I'm my case, I'm using Kitty's keybindings to trigger opening in CWD:

```
map kitty_mod+enter new_window_with_cwd
map kitty_mod+t new_tab_with_cwd
```

Any suggestions for how to further debug this? (I guess I could start a git bisect...)

**Comment by kovidgoyal** [comment]
I dont really see how this could be related to kitty, all kitty has to do in this scenario is set the cwd in the child process before execing the shell. It is obviously doing that, since ls and frieds work. So why pwd does not in fish, is a mystery.

But feel free to bisect, may turn up something unexpected.

**Comment by callahad** [comment]
Looks like an upstream issue in Fish; downgraded to 2.7.1 and the `with_cwd` commands work as expected.

**Comment by callahad** [comment]
From discussion at https://github.com/fish-shell/fish-shell/issues/5647, this should be resolved in Fish 3.0.2.

From local testing, that appears to be the case.
